ethereum-optimism/optimism
# Develop -> Master

- op-e2e: Add e2e tests for bedrock behaviour that will change in regolith
- Remove SourceHash which was being calculated incorrectly
- op-node: Fix storage slot comment
- op-chain-ops: Do not manually copy balance in WipePredeployStorage
- Increase `FeeVault` withdrawal tx min gas limit
- contracts-bedrock: remove crossdomain messenger pausability
- op-chain-ops: remove dead storage values
- op-chain-ops: update initialize call with no arg
- contracts-bedrock: fix checkl2
- contracts-bedrock: fix deployment
- ctb: refactor layout script
- op-node: handle sequencer errors with derivation-levels, reset when L1 origins are inconsistent, test reset handling
- op-node: fix-sequencer-err-handling PR 4930 suggestions
- fix(specs): Corrections to messenger workflow and iface
- contracts-bedrock: slight cleanup
- op-node: fix redundant checks, check id before cancellation, add reset upon sealing reset-err, extend godoc
- Update op-node/withdrawals/utils.go
- specs: define Regolith upgrade changes
- build(deps): bump golang.org/x/net in /indexer
- feat(bindings): Print abigen version in CI
- chore: Add .editorconfig settings for Makefile
- ci: Combine release jobs, add release approval
- build(deps): bump golang.org/x/sys in /proxyd
- build(deps): bump golang.org/x/sys in /l2geth-exporter
- build(deps): bump golang.org/x/sys in /gas-oracle
- ci: Add tag versions to Docker binaries
- ci: Modify approval job to run on op-* branches, remove context
- ci: Upgrade GCP CLI orb
- ci: Remove errant space
- fix(contracts-periphery): Fix attestation station test
- Add eth denver logo
- Revert change I made to wrong repo
- Create olive-wasps-love.md
- Create serious-tools-try.md
- specs: regolith deposit spec review-changes
- build(deps): bump golang.org/x/crypto
- build(deps): bump golang.org/x/crypto in /l2geth-exporter
- Create gentle-mangos-travel.md
- fix(ops): Fix wd-mon name in Dockerfile.packages
- ci: Add wd-mon to release workflow
- build(deps): bump golang.org/x/net in /gas-oracle
- build(deps): bump golang.org/x/crypto in /proxyd
- Create healthy-fans-dance.md
- build(deps): bump golang.org/x/net in /op-exporter
- build(deps): bump golang.org/x/crypto in /batch-submitter
- Create tidy-ads-smell.md
- op-geth: Update to latest op-geth
- op-e2e, op-chain-ops/genesis: Add e2e op-geth tests for Regolith changes
- op-chain-ops/genesis: Enable Regolith as part of initial migration
- contracts-bedrock: use `1 gwei` over hardcoded number
- contracts-bedrock: ResourceMetering more readable
- Add support for stopping / starting the batcher
- Minor formatting
- clarified comment
- :sparkles: Add atst boilerplate
- :sparkles: Add atst SDK
- move DEFAULT_DATA_TYPE based on comment suggestion
- attestations too
- Add `ResolvedDelegateProxy` tests
- Expect the delegatecalls
- Logging fixes
- Lint
- More logging fixes
- :sparkles: Add atst CLI
- fix write
- comments
- Add E2E test
- Ignore jetbrains config
- Import sort
- Fix tests
- op-e2e: Update TestCrossLayerUser test to check deposits and tx work with and without regolith
- Install correct node version before yarn install
- documentation
- export stringifyAttestationBytes
- correctly make ethers a peer dep
- add sdk docs
- add the cli docs
- fix readme missing ethers dep
- update react
- update logo
- add npx to cli docs
- add react hooks
- fix 0x bug
- changeset
- Update CONTRIBUTING.md
- Apply suggestions from code review
- op-node/p2p: Fix flaky TestDiscovery
- chore(cmon): Add a changeset
- Do not allow the challenger to delete finalized outputs
- Move `FINALIZATION_PERIOD_SECONDS` to `L2OutputOracle`; Need to update deployment etc.
- Deployment + binding; go changes now
- Chain ops changes
- SDK changes
- Add changeset
- Mark's nits
- Lint
- SDK release minor -> major
- Semver bump on the portal + oracle
- make the types better
- update parseFoo
- changeset
- delete unused link
- nit
- Update README.md
- add encodeRawKey
- Update packages/atst/docs/sdk.md
- fix linter
- add eventFilters
- op-e2e: Properly run TestMixedWithdrawalValidity as parallel
- ci: Fix gotestum timeout
- op-batcher: Make ChannelBuilder.InputTargetReached private
- op-batcher: Add channel duration tracking to the channel builder
- op-batcher,e2e: Add MaxChannelDuration parameter to batcher
- op-batcher: Move L1 block registration after L2 blocks processing
- op-e2e: Increase timeouts in tests that wait for finalization
- op-batcher: Improve explanation of MaxChannelDuration parameter
- fix(atst): return correct type for encodeRawKey
- fix(atst): Return right type for encodeRawKey
- changset
- Update .changeset/nasty-chairs-agree.md
- hexlify
- fix bug with 0x0
- update cli and missing import
- - Add ecopod as codeowner to @eth-optimism/atst
